[SDCI-666] Add docs manual correlation for ArgoCD deployments (#26028)
DataDog/documentation: content/en/continuous_delivery/deployments/argocd.md

{{< img src="ci/diagram_argo-cd-deployment_240910.png" alt="Triggering Argo CD deployments using git" style="width:100%;">}}

> The `datadog-ci deployment correlate` command can be used to correlate one or more configuration repository commits with an application repository commit. When an Argo CD deployment occurs, the configuration commit information in the deployment event is replaced by the related application repository commit, if any. There are two possible ways to perform the correlation using the command: automatic and manual setup. Both methods require version `2.44.0` or higher of the `datadog-ci` CLI.

Reviewer: Can we add a link for datadog-ci similar to other places?

{{< img src="ci/diagram_argo-cd-deployment_240910.png" alt="Triggering Argo CD deployments using git" style="width:100%;">}}

> The `datadog-ci deployment correlate` command can be used to correlate one or more configuration repository commits with an application repository commit. When an Argo CD deployment occurs, the configuration commit information in the deployment event is replaced by the related application repository commit, if any. There are two possible ways to perform the correlation using the command: automatic and manual setup. The version of the `datadog-ci` CLI should be `2.44.0` or higher.

Reviewer 2: ```suggestion
The `datadog-ci deployment correlate` command can be used to correlate one or more configuration repository commits with an application repository commit. When an Argo CD deployment occurs, the configuration commit information in the deployment event is replaced by the related application repository commit, if any. There are two possible ways to perform the correlation using the command: automatic and manual setup. Both methods require version `2.44.0` or higher of the `datadog-ci` CLI.
```

{{< tabs >}}
{{% tab "Automatic" %}}

> When this approach is used, the command automatically infers the current application commit (that is, the commit of the pipeline the command is running in) and the configuration repository commits based on the current Git environment. For this to work properly, the command needs to be run between committing the changes and pushing them to the configuration repository:

Reviewer 2: ```suggestion
In this method, the command automatically infers the current application commit (that is, the commit of the pipeline that the command is running in) and the configuration repository commits based on the current Git environment. For this to work properly, the command needs to be run between committing the changes and pushing them to the configuration repository:
```

In this case, you can replace the Git metadata reported in the deployment with the metadata of the application repository instead of the configuration repository. This allows you to connect the deployments performed by Argo CD and the related CI pipeline runs on the application repository.

To associate the application repository Git information with the Argo CD deployments, run the `datadog-ci deployment correlate` in your CI. The required command arguments vary by the setup:
* **Direct push**: If the CI pipeline directly pushes the changes to the configuration repository and triggers an Argo CD deployment.

> * **Manually**: If the CI pipeline does not directly push the changes to the configuration repository. For example: the CI creates a feature branch, commits to it and lastly merges it.

Author: For `1`, I would not say it only works with the default branch only. For example, you can push your changes to a feature branch and If your ArgoCD application monitors feature branches, it will trigger a deployment and the correlation will work.

    datadog-ci deployment correlate --provider argocd
    git push
```
{{% /tab %}}
{{% tab "Manual" %}}

> If the automatic setup is too limited for your use case, it is possible to provide the the configuration repository URL and SHAs manually:

Reviewer: ```suggestion
If the automatic setup is too limited for your use case, it is possible to provide the configuration repository URL and SHAs manually:
```

```yaml
- job: JobToUpdateConfigurationRepository
  run: |
    # Correlate the deployment with the CI pipeline
    export DD_BETA_COMMANDS_ENABLED=1
    datadog-ci deployment correlate --provider argocd --config-repo <CONFIG_REPO_URL> --config-shas <COMMIT_SHA>
```

> You can omit the `--config-repo` option if the CI is checked out to the configuration repository. See [command syntax][14] for additional details.

Reviewer: The link is not working from the preview

    git push
```
{{% /tab %}}
{{% tab "Manual" %}}

> If the automatic setup is too limited for your use case, it is possible to provide the configuration repository URL and SHAs manually:

Reviewer 2: ```suggestion
If the automatic setup is too limited for your use case, you can provide the configuration repository URL and SHAs manually:
```

```yaml
- job: JobToUpdateConfigurationRepository
  run: |
    # Correlate the deployment with the CI pipeline
    export DD_BETA_COMMANDS_ENABLED=1
    datadog-ci deployment correlate --provider argocd --config-repo <CONFIG_REPO_URL> --config-shas <COMMIT_SHA>
```
You can omit the `--config-repo` option if the CI is checked out to the configuration repository. See [command syntax][14] for additional details.

> {{% /tab %}}

Reviewer 2: ```suggestion
You can omit the `--config-repo` option if the CI is checked out to the configuration repository. See [command syntax][1] for additional details.

[1]: https://github.com/DataDog/datadog-ci/tree/master/src/commands/deployment#correlate
{{% /tab %}}
```
